src/main/java/org/example/dao/PointDAOImpl.java

```diff
@@ -0,0 +1,104 @@
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public PointDAOImpl() {
+    }
+
+    @Override
+    public void insertPoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO point (id, xCoordinate, yCoordinate) VALUES (?, ?, ?)", Statement.RETURN_GENERATED_KEYS);
+            statement.setInt(1, point.getId());
+            statement.setDouble(2, point.getXCoordinate());
+            statement.setDouble(3, point.getYCoordinate());
+            statement.executeUpdate();
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                int id = generatedKeys.getInt(1);
+                point.setId(id);
+                System.out.println("Point inserted successfully. Generated ID: " + id);
+            }
+
+            statement.close();
+            generatedKeys.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public void deletePoint(Point point) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        try {
+            PreparedStatement statement = connection.prepareStatement("DELETE FROM point WHERE id = ?");
+            statement.setInt(1, point.getId());
+            statement.executeUpdate();
+            System.out.println("Point deleted successfully.");
+            statement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+    }
+
+    @Override
+    public Point getPoint(int id) {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        Point point = null;
+        try {
+            PreparedStatement statement = connection.prepareStatement("SELECT * FROM point WHERE id = ?");
+            statement.setInt(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                double xCoordinate = resultSet.getDouble("xCoordinate");
+                double yCoordinate = resultSet.getDouble("yCoordinate");
+                point = new Point(id, xCoordinate, yCoordinate);
+            }
+            statement.close();
+            resultSet.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }finally {
+            CONNECTIONPOOL.releaseConnectionToPool(connection);
+        }
+        return point;
+    }
+
+    @Override
+    public List<Point> getPoints() {
+        Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+        List<Point> points = new ArrayList<>();
+        try {
+            Statement statement = connection.createStatement();
```

Comment:
It may be better to use the PreparedStatement instead of Statement here like we do in all other methods.

Reply:
changed